src/OmniSharp.Roslyn.CSharp/Services/Diagnostics/CodeCheckService.cs

```diff
@@ -13,22 +13,47 @@ namespace OmniSharp.Roslyn.CSharp.Services.Diagnostics
     [OmniSharpHandler(OmniSharpEndpoints.CodeCheck, LanguageNames.CSharp)]
     public class CodeCheckService : IRequestHandler<CodeCheckRequest, QuickFixResponse>
     {
-        private OmniSharpWorkspace _workspace;
+        private readonly OmniSharpWorkspace _workspace;
+        private readonly RoslynAnalyzerService _roslynAnalyzer;
+        private readonly ILogger<CodeCheckService> _logger;
 
         [ImportingConstructor]
-        public CodeCheckService(OmniSharpWorkspace workspace)
+        public CodeCheckService(OmniSharpWorkspace workspace, RoslynAnalyzerService roslynAnalyzer, ILoggerFactory loggerFactory)
```

Comment:
Small bit of trivia, `CodeCheckService` will eventually go away, it's kind of expensive to go and find all the diagnostics for a solution (especially large ones) the `DiagnosticsService` was added to replace code check eventually, with a pushed instead of pull based model.  

Currently DiagnosticService is only used by LSP, but it is the default way to present diagnostics in the UI for LSP itself.

Reply:
`DiagnosticService` however needs some TLC to make a little smarter about getting documents for the entire solution, sometimes it misses diagnostics if the class changes in one file.